server/src/controllers/course.ts

```diff
@@ -90,19 +101,105 @@ function parseSisuInstance(instance: SisuInstance): InstanceData {
   };
 }
 
+// Middleware function for handling errors in JSON parsing
+export function handleInvalidRequestJson(err: JsonError, req: Request, res: Response, next: NextFunction): void {
+  if (err instanceof SyntaxError && err.status === 400 && err.body !== undefined) {
+    res.status(400).send({
+      success: false,
+      errors: [ `SyntaxError: ${err.message}: ${err.body}` ]
+    });
+  } else
+    next();
+}
 
 export async function addCourse(req: Request, res: Response): Promise<void> {
+  const requestSchema: yup.AnyObjectSchema = yup.object().shape({
+    courseCode: yup.string().required(),
+    translations: yup.array().of(yup.object().shape({
+      lang: yup.mixed<Language>().oneOf(Object.values(Language)).required(),
+      department: yup.string().required(),
+      name: yup.string().required()
+    })).required()
+  });
+
+  const t: Transaction = await sequelize.transaction();
```

Comment:
Remember to pass the [transaction](https://sequelize.org/docs/v6/other-topics/transactions/#unmanaged-transactions) to those db calls you want to use it. Now none of the calls utilize this and commit/rollback will do nothing.

Reply:
Oh fuck, true... Will fix this.

Reply:
We probably have to figure out a more thorough testing solution for db operations in future. Not directly concerning this code snippet though.